include/irrMath.h

```diff
@@ -212,14 +186,20 @@ namespace core
 		return 8;
 	}
 
+	//! returns if a equals b, for types without rounding errors
+	template <class T, std::enable_if_t<std::is_integral<T>::value, bool> = true>
+	inline bool equals(const T a, const T b)
+	{
+		return a == b;
+	}
+
 	//! returns if a equals b, taking possible rounding errors into account
-	template <class T>
+	template <class T, std::enable_if_t<std::is_floating_point<T>::value, bool> = true>
 	inline bool equals(const T a, const T b, const T tolerance = roundingError<T>())
 	{
-		return (a + tolerance >= b) && (a - tolerance <= b);
+		return abs(a - b) <= tolerance;
```

Comment:
Is this supposed to be used for floating points?
The stdlib signature of abs is `int abs(int x)`, so this doesn't make sense.

Not sure why, but MSVC generates an ambiguity error here in my xr branch but not master (possibly because I've got C++20 enabled)

Reply:
it's defined for different float and int types:
* https://en.cppreference.com/w/cpp/numeric/math/fabs
* https://en.cppreference.com/w/cpp/numeric/math/abs

Reply:
Ah. It needs to be std::abs then to avoid picking up the integer version. And the includes need to be cstdio and cmath.

Reply:
Can you send a PR?

Reply:
sent: https://github.com/minetest/irrlicht/pull/293